third_party/terraform/resources/resource_composer_environment.go.erb

```diff
@@ -923,3 +927,61 @@ func validateServiceAccountRelativeNameOrEmail(v interface{}, k string) (ws []st
 
 	return
 }
+
+<% unless version.nil? || version == 'ga' -%>
+func composerImageVersionDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
+	versionRe := regexp.MustCompile(composerEnvironmentVersionRegexp)
+	oldVersions := versionRe.FindStringSubmatch(old)
+	newVersions := versionRe.FindStringSubmatch(new)
+	if oldVersions == nil || len(oldVersions) < 3 {
+		// Somehow one of the versions didn't match the regexp or didn't
+		// have values in the capturing groups. In that case, fall back to
+		// an equality check.
+		log.Printf("[WARN] Composer version didn't match regexp: %s", old)
+		return old == new
+	}
+	if newVersions == nil || len(newVersions) < 3 {
+		// Somehow one of the versions didn't match the regexp or didn't
+		// have values in the capturing groups. In that case, fall back to
+		// an equality check.
+		log.Printf("[WARN] Composer version didn't match regexp: %s", new)
+		return old == new
+	}
+
+	// Check airflow version using the version package to account for
+	// diffs like 1.10 and 1.10.0
+	eq, err := versionsEqual(oldVersions[2], newVersions[2])
+	if err != nil {
+		log.Printf("[WARN] Could not parse airflow version, %s", err)
+	}
+	if !eq {
+		return false
+	}
+
+	// Check composer version. Assume that "latest" means we should
```

Comment:
I'd prefer requiring users to use a datasource to specify latest versions, I think? We can't diff here, so the config will read `"latest'` even if the user's version is sorely outdated. It's a worse UX in the simple case but it means that Terraform will work correctly, producing a diff when there is a new latest version.

Reply:
Or, I bet we could use a `CustomizeDiff` to expand `latest` to the real latest version if we want to preserve the better UX.

Reply:
The field is ForceNew though, so if we showed a diff when there was a new latest version then people's environments would get destroyed. If they do want to destroy/create, they can taint the resource.

Also, there isn't an API for versions so we don't actually have a way of knowing what version the latest one is besides actually spinning up an environment and checking, which seems very wasteful.

Reply:
Ah, that's sad. Fair enough.